WPMedia/arc-themes-components
# Tmedia 671 caption component

Ticket: https://arcpublishing.atlassian.net/jira/software/c/projects/TMEDIA/boards/875?modal=detail&selectedIssue=TMEDIA-671&assignee=5e387d6a1b1d910e5dfd7a9b

# Ac: 

Create a <Caption> component

Uses props of Title, Caption and Credit that can be hidden at the block level

May have HTML in it which should render (i.e. `<b>, <i>`)

Have Scss file that adds the hooks to the Sass library

Includes three component properties: Title, Caption, and Credit

Caption - Title

Caption - Caption

Caption - Credit

Create a utility function to handle formatting credits that can be used by consumers - but not built into this component - [engine-theme-sdk/index.tsx at arc-themes-release-version-1.20 · WPMedia/engine-theme-sdk](https://github.com/WPMedia/engine-theme-sdk/blob/arc-themes-release-version-1.20/src/components/ImageMetadata/index.tsx#L47) 

Have test coverage

Have storybook documentation

# Test: 

1. Run `npm run storybook`
2. Go to Caption component 
3. See effect of html passed in via story `Renders html passed into title or caption` 
4. Look at how credit can take in multiple affiliations and/or bylines with stories


## Feedback 

- @nschubach's suggestion of using `.toHaveClass` was not doable as there was no way for me to select figcaption without role or without any text https://github.com/WPMedia/arc-themes-components/pull/29#pullrequestreview-896923800
- Removed out of scope sanitize if security is not a concern https://github.com/WPMedia/arc-themes-components/pull/29#discussion_r819742123
- Used dangerously set html based on feedback and requirements https://github.com/WPMedia/arc-themes-components/pull/29#discussion_r817198643
- Naming standards discussion from before
